runtime/src/main/java/org/corfudb/protocols/wireprotocol/TailsResponse.java

```diff
@@ -0,0 +1,49 @@
+package org.corfudb.protocols.wireprotocol;
+
+import io.netty.buffer.ByteBuf;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ *
+ * A container that contains information about the log's stream tails
+ * map and max global address written (i.e. global tail)
+ *
+ * Created by Maithem on 10/22/18.
+ */
+
+@Data
+@RequiredArgsConstructor
+public class TailsResponse implements ICorfuPayload<TailsResponse> {
+
+    final long logTail;
+
+    final Map<UUID, Long> streamTails;
+
+    public TailsResponse(ByteBuf buf) {
+        streamTails = new HashMap<>();
```

Comment:
You can use the existing serializers and de-serializers. It can serialize UUIDs.

Reply:
done.